### PR TITLE
Required annotation implement

### PR DESCRIPTION
**Pull request by threat70**

### CODE REVIEW

**Comment by threat70** [comment]
I signed it!

**Comment by threat70** [comment]
I signed it!

File: README.md [hunk 1]
```diff
@@ -1,26 +1,13 @@
...
-  * Gson [user guide](http://sites.google.com/site/gson/gson-user-guide): This guide contains examples on how to use Gson in your code.
-  * Gson [Roadmap](https://sites.google.com/site/gson/gson-roadmap): Details on upcoming releases 
-  * Gson [design document](https://sites.google.com/site/gson/gson-design-document): This document discusses issues we faced while designing Gson. It also include a comparison of Gson with other Java libraries that can be used for Json conversion
-
-Please use the [google-gson Google group](http://groups.google.com/group/google-gson) to discuss Gson, or to post questions. 
-
-*Gson-related Content Created by Third Parties*
-  * [Gson Tutorial](http://www.studytrails.com/java/json/java-google-json-introduction.jsp) by `StudyTrails`
+# google-gson with @Required annotation
+
```
Comment (inder123) [inline README.md]: Why will you delete existing documentation?

**Comment by inder123** [comment]
Seems interesting, however I think we should have a more general way of validating an object post deserialization. @swankjesse  @JakeWharton @joel-leitch  Any thoughts?

**Comment by threat70** [comment]
sorry, my mistake. copied from fork

**Comment by swankjesse** [comment]
Yeah, if we're going to validate that all required fields are present, we probably also want to support arbitrary validation, and that will require careful & thoughtful API design. What do other JSON libraries do?

**Comment by inder123** [comment]
Closing this pull request, as we will opt for a more general validation design. Thank you for your work @threat70. Appreciate the fact that you took the time to write unit tests as well.

**Pull request closed without merging**